larpandoracontent/LArThreeDReco/LArCosmicRay/NViewDeltaRayMatchingAlgorithm.cc

```diff
@@ -0,0 +1,1105 @@
+
+    xOverlapObject = XOverlap(xMinU, xMaxU, xMinV, xMaxV, xMinW, xMaxW, xCentreOverlap);
+
+    return STATUS_CODE_SUCCESS;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+template<typename T>
+void NViewDeltaRayMatchingAlgorithm<T>::GetClusterSpanX(const CaloHitList &caloHitList, float &xMin, float &xMax) const
+{
+    xMin = std::numeric_limits<float>::max();
+    xMax = -std::numeric_limits<float>::max();
+    
+    for(const CaloHit *const pCaloHit : caloHitList)
+    {
+        const float xCoordinate(pCaloHit->GetPositionVector().GetX());
+
+        if (xCoordinate < xMin)
+            xMin = xCoordinate;
+
+        if (xCoordinate > xMax)
+            xMax = xCoordinate;
+    }
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+template<typename T>
+StatusCode NViewDeltaRayMatchingAlgorithm<T>::GetClusterSpanZ(const CaloHitList &caloHitList, const float xMin, const float xMax, float &zMin, float &zMax) const
+{
+    zMin = std::numeric_limits<float>::max();
+    zMax = -std::numeric_limits<float>::max();
+
+    bool found(false);
+    for (const CaloHit *const pCaloHit : caloHitList)
+    {
+        const float xCoordinate(pCaloHit->GetPositionVector().GetX());
+        const float zCoordinate(pCaloHit->GetPositionVector().GetZ());
+        
+        if ((xCoordinate < xMin) || (xCoordinate > xMax))
+            continue;
+
+        found = true;
+
+        if (zCoordinate < zMin)
+            zMin = zCoordinate;
+
+        if (zCoordinate > zMax)
+            zMax = zCoordinate;
+    }
+
+    if (!found)
+        throw StatusCodeException(STATUS_CODE_NOT_FOUND);
+
+    return STATUS_CODE_SUCCESS;
+}
+
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+template<typename T>
+StatusCode NViewDeltaRayMatchingAlgorithm<T>::ProjectMuonPositions(const HitType &thirdViewHitType, const ParticleFlowObject *const pParentMuon,
+    CartesianPointVector &projectedPositions) const
+{
+    ClusterList muonClusterList1, muonClusterList2;    
+
+    for (const HitType hitType1 : {TPC_VIEW_U, TPC_VIEW_V, TPC_VIEW_W})
+    {
+        if (hitType1 == thirdViewHitType)
+            continue;
+        
+        for (const HitType hitType2 : {TPC_VIEW_U, TPC_VIEW_V, TPC_VIEW_W})
+        {
+            if ((hitType2 == thirdViewHitType) || (hitType1 == hitType2))
+                continue;
+
+            LArPfoHelper::GetClusters(pParentMuon, hitType1, muonClusterList1);
+            LArPfoHelper::GetClusters(pParentMuon, hitType2, muonClusterList2);
+            
+            if ((muonClusterList1.size() != 1) || (muonClusterList2.size() != 1))
+                return STATUS_CODE_NOT_FOUND;
+            
+            break;
+        }
+
+        break;
+    }
```

Comment:
I think this can use a similar approach to that suggested here: https://github.com/PandoraPFA/LArContent/pull/165#discussion_r616715607